src/garage/tf/policies/base.py

```diff
@@ -23,41 +23,68 @@ def __init__(self, name, env_spec):
         self._cached_param_shapes = None
 
     @abc.abstractmethod
-    def get_action(self, observation):
+    def get_action(self, observation, agent_infos):
```

Comment:
we should take this opportunity to make sure `agent_infos` is the right name for this stuff.

i suggest you rename it to `policy_info`(s) because we don't have anything in `garage` named `Agent` and probably won't in the future. if someone comes across `policy_info` in a debugger they have a good chance of guessing it came from a `Policy` but if they come across `agent_info` they won't find anything named `agent` which is unnecessarily obtuse.

please leave a note in the policy interface which makes it clear the purpose of `policy_info` and what people should and should not do with it. basically it will be something like


Note:
`policy_info` is intended to allow policies to pass state to themselves between calls to `get_actions`. Users should consider its contents to be private, and should never assume this variable contains anything in particular, or even that its contents is consistent between calls to the same `Policy` object. User should *never* mutate the contents of `policy_info` and no `Policy` should ever require the user inspect or mutate `policy_info` as part of its interface. Implementers of `Policy` may occasionally find it useful to store some debugging information in `policy_info`, but any information implementers would like to be consistently logged by users should be exposed explicitly through the `get_diagnostics` interface.

Reply:
fixed